packages/framework/src/Model/Product/Search/Export/ProductSearchExportWithFilterRepository.php

```diff
@@ -83,50 +83,28 @@ public function __construct(
     /**
      * @param int $domainId
      * @param string $locale
-     * @param int $startFrom
+     * @param int $lastProcessedId
      * @param int $batchSize
      * @return array
      */
-    public function getProductsData(int $domainId, string $locale, int $startFrom, int $batchSize): array
+    public function getProductsData(int $domainId, string $locale, int $lastProcessedId, int $batchSize): array
     {
         $queryBuilder = $this->createQueryBuilder($domainId)
-            ->setFirstResult($startFrom)
+            ->andWhere('p.id > :lastProcessedId')
+            ->setParameter('lastProcessedId', $lastProcessedId)
             ->setMaxResults($batchSize);
 
         $query = $queryBuilder->getQuery();
 
         $results = [];
         /** @var \Shopsys\FrameworkBundle\Model\Product\Product $product */
         foreach ($query->getResult() as $product) {
-            $results[] = $this->extractResult($product, $domainId, $locale);
+            $results[$product->getId()] = $this->extractResult($product, $domainId, $locale);
         }
 
         return $results;
     }
```

Comment:
maybe it's time to rename `ProductSearchExportWithFilter` to something better, what do you think?

Reply:
It is not necessary to solve this now.